contrib/systemd/systemd.py

```diff
@@ -0,0 +1,133 @@
+    while True:
+        dst_qs = pipe_data[cbot].get('destination-queues', [])
+        if len(dst_qs) == 1:
+            dst_q = dst_qs[0]
+            _bot = sqs2bot[dst_q]
+            cbot = _bot[0]
+            count = dsq_count[dst_q]
+            if count > 1:
+                break
+            else:
+                cbs.append(cbot)
+        else:
+            break
+    return cbs
+
+
+def main():
+    with open(RUNTIME_CONF_FILE, encoding='utf-8') as rc_file:
+        rc_data = json.loads(rc_file.read())
+
+    with open(PIPELINE_CONF_FILE, encoding='utf-8') as pipe_file:
+        pipe_data = json.loads(pipe_file.read())
+
+    if not os.path.exists(SYSTEMD_OUTPUT_DIR):
+        os.makedirs(SYSTEMD_OUTPUT_DIR)
+
+    collectors = [i for i in rc_data if rc_data[i]['group'] == 'Collector']
+    # parsers = [i for i in pipe_data if rc_data[i]['group'] == 'Parser']
+
+    for bot in collectors:
+        bot_data = rc_data[bot]
+
+        if DISABLE_IN_CONF:
+            rc_data[bot]['enabled'] = False
+
+        if SET_RUNMODE_IN_CONF:
+            rc_data[bot]['run_mode'] = 'scheduled'
+
+        bot_run_cmd = INTELMQCTL_BIN + ' run ' + bot
+        service_file_name = os.path.join(SYSTEMD_OUTPUT_DIR,
+                                         SERVICE_PREFIX + bot + '.service')
+        service_data = SERVICE_TEMPLATE.substitute(INTELMQ_USER=INTELMQ_USER,
+                                                   INTELMQ_GROUP=INTELMQ_GROUP,
+                                                   bot=bot,
+                                                   bot_run_cmd=bot_run_cmd
+                                                    )
+        with open(service_file_name, "w", encoding='utf-8') as svc_file:
+            svc_file.write(service_data)
+
+        bot_parameters = bot_data['parameters']
+        bot_service_name = SERVICE_PREFIX + bot + '.service'
+        bot_interval = int(bot_parameters['rate_limit'])
+        timer_file_name = os.path.join(SYSTEMD_OUTPUT_DIR,
+                                       SERVICE_PREFIX + bot + '.timer')
+
+        timer_data = TIMER_TEMPLATE.substitute(bot=bot,
+                                               ACCURACY_SECS=ACCURACY_SECS,
+                                               RANDOMIZE_DELAYS=RANDOMIZE_DELAYS,
+                                               ON_ACTIVE_SEC=ON_ACTIVE_SEC,
+                                               bot_interval=bot_interval,
+                                               bot_service_name=bot_service_name
+                                               )
+        with open(timer_file_name, "w", encoding='utf-8') as tmr_file:
+            tmr_file.write(timer_data)
+
+    if DISABLE_IN_CONF or SET_RUNMODE_IN_CONF:
+        shutil.move(RUNTIME_CONF_FILE, RUNTIME_CONF_FILE + '.bak')
+        rc_data = collections.OrderedDict(sorted(rc_data.items()))
+        data = json.dumps(rc_data, indent=4)
+        with open(RUNTIME_CONF_FILE, "w", encoding='utf-8') as rc_file:
```

Comment:
You can reuse intelmqctl's [`write_updated_runtime_config`](https://github.com/certtools/intelmq/blob/bc272d1f7ca68107ad6e0313728c051b7cf76add/intelmq/bin/intelmqctl.py#L611) here.